backend/src/main/java/heartbeat/service/pipeline/buildkite/BuildKiteService.java

```diff
@@ -191,4 +194,39 @@ private int parseTotalPage(@Nullable List<String> linkHeader) {
 		return 1;
 	}
 
+	public List<BuildKiteBuildInfo> fetchPipelineBuilds(String token, DeploymentEnvironment deploymentEnvironment,
+			String startTime, String endTime) {
+		String partialToken = token.substring(0, token.length() / 2);
+		PipelineStepsParam stepsParam = new PipelineStepsParam(startTime, endTime);
+
+		return this.fetchPipelineStepsByPage(token, deploymentEnvironment.getOrgId(), deploymentEnvironment.getId(),
+				stepsParam, partialToken);
+	}
+
+	public DeployTimes countDeployTimes(DeploymentEnvironment deploymentEnvironment,
+			List<BuildKiteBuildInfo> buildInfos) {
+		if (deploymentEnvironment.getOrgId() == null) {
+			throw new Error("miss orgId argument");
```

Comment:
Change with the Codacy suggestion

Reply:
Define a NotFoundException class to catch this error.